src/translator/service.h

```diff
@@ -108,15 +109,26 @@ class BlockingService {
 class AsyncService {
  public:
   struct Config {
-    size_t numWorkers{1};   ///< How many worker translation threads to spawn.
-    size_t cacheSize{0};    ///< Size in History items to be stored in the cache. Loosely corresponds to sentences to
-                            /// cache in the real world. A value of 0 means no caching.
+    std::vector<size_t> gpuWorkers;  ///< GPU workers array. If not-empty use CPU workers instead.
+    size_t numWorkers{1};            ///< How many worker translation threads to spawn.
+    size_t cacheSize{0};  ///< Size in History items to be stored in the cache. Loosely corresponds to sentences to
+                          /// cache in the real world. A value of 0 means no caching.
+    std::string terminologyFile{""};
+    bool terminologyForce{false};
     Logger::Config logger;  // Configurations for logging
+    std::string format{"%s __target__ %s __done__ "};
 
     template <class App>
     static void addOptions(App &app, Config &config) {
       app.add_option("--cpu-threads", config.numWorkers, "Workers to form translation backend");
+      app.add_option("--gpu-workers", config.gpuWorkers, "GPU workers for the translation backend.");
       app.add_option("--cache-size", config.cacheSize, "Number of entries to store in cache.");
+      app.add_option("--terminology-file", config.terminologyFile, "tsv, one term at a time terminology file.");
+      app.add_option(
+          "--force-terminology", config.terminologyForce,
+          "Force the terminology to appear on the target side. May degrade translation quality. Not recommended.");
+      app.add_option("--terminology-form", config.format,
+                     "Form for technology. Default is \"%s __target__ %s __done__ \". Change depending on the model.");
```

Comment:
Do we want to figure out how to store this in the model yaml instead before we merge?

Reply:
Pedantically, we probably should.